### ISSUE TITLE
Add qubes-qvm-screenshot-tool dom0 salt recipe deployment

### ISSUE THREAD

**Issue by tlaurion** [ben-grande/qusal#22]
### Commitment

I confirm that I have read the following resources:

- [How to ask questions The Smart Way](http://catb.org/esr/faqs/smart-questions.html)
- [Writing the perfect question](https://codeblog.jonskeet.uk/2010/08/29/writing-the-perfect-question/)
- [Question checklist](https://codeblog.jonskeet.uk/2012/11/24/stack-overflow-question-checklist/)
- [Could you please make my preference the default?](https://www.qubes-os.org/faq/#could-you-please-make-my-preference-the-default)

### Current problem (if any)

Dom0 screenshot and setting preferences shortcuts is not so straightforward for end users. One amazing project was put under contrib packages, but bootstrapping first package installation is also not straightforward for end users. 

### Proposed solution

It would be really helpful to offer dom0 installation of  https://github.com/QubesOS-contrib/qubes-qvm-screenshot-tool with a dom0 salt recipe, binding xfce alt-printscreen and printscreen keyboard shortcuts to both KDE and Xfce. 

### The value to a user, and who that user might be

Sharing a windows content or whole desktop through copying it directly to proper domu is already properly resolved by https://github.com/QubesOS-contrib/qubes-qvm-screenshot-tool. On system reinstallation, it should be as easy to deploy as qusal makes it for other tools. 

Of course this is my preference. Not sure why this is not the default yet and in contrib repo, but that tool is available and superior to the default. Trying it is adopting it, with space for improvement upstream for it to eventually become the default.

**Comment by ben-grande** [comment]
I rewrite every script added to Qusal to follow some coding rules and I as far as I can see, I will need to rewrite that tool completely. Can you please inform what features you use from that tool or if you know someone that uses it, what they most need?

- Is `hosting on imgurl` really necessary? I am not okay with automatically hosting on any service, it also requires `curl` installed and network connectivity.
- Is `opening file manager` commonly used? I am ok with this part.
- Is `copy image to clipboard` commonly used? I am ok with this but it expects the target qube to have `xclip` or maybe `xsel` installed.

I don't commonly do snapshots but from seeing forum posts, I can see it is very common. I can see how a 3 step solution:

1. screenshot
2. qvm-copy-to-vm
3. open nautilus or saved to clibpoard

Can be converted to a single option to export to a qube, better usability.

> Of course this is my preference. Not sure why this is not the default yet and in contrib repo, but that tool is available and superior to the default. Trying it is adopting it, with space for improvement upstream for it to eventually become the default.

If you just want Qusal to bootstrap by [enabling the contrib repo and downloading the tool from the contrib repo](https://www.qubes-os.org/doc/installing-contributed-packages/), although that is possible, I am still thinking towards writing my own version. I know these rewrites takes me some time but the end result is much better, in my point of view, of course. This, of course, depends on how you answer the questions above.

**Comment by tlaurion** [comment]
@ben-grande : I think an iterative approach would be delightful here
1. Bootstrap the contrib repos install, key deployment and repo url changes if cacher installed
2. Deploy qubes-qvm-screenshot
3. Modify qubes-qvm-screenshot

Base logic of this is, if I understand contribution and upstreamont correctly, contrib is first step prior of adoption under main repos.


The main reason I use it is for print screen and alt-printscreen replacement, that is, to only select part of the screen related to a qube window prior of having the tool ask what to do next. Nor ally that is simply to call the qvm-copy to desired vm and that's it.

I agree that editing with imagemagik is less than efficient, but I'm not aware of the other use cases like paste bin. Afterall, if final screenshot is in desired vm, you can then edit it if needed or add text etc from better tools there.

The main thing here is really to have contrib repos activated alongside all qusal helpers (cacher) and then deploy that tool as is + keyboard shortcut remappings (xfce and KDE).


Makes sense?

**Comment by ben-grande** [comment]
Yes, it makes sense. I started doing the tool in the meantime because I didn't want to add a tool from the contrib repository that has unnecessary stuff.

> Base logic of this is, if I understand contribution and upstreaming correctly, contrib is first step prior of adoption under main repos.

Most things I see in contrib are never added to the main repos. This is why most things I did contribute to Qubes either make to the default installation or don't make to contrib at all, as I don't see a value in that due to low reviews and actions.

- [where "python" modules go to die](https://leancrew.com/all-this/2012/04/where-modules-go-to-die/)
- [my comment about contributed packages](https://github.com/QubesOS/qubes-issues/issues/1939#issuecomment-1896477279)

So, I either wait for upstream to have enough time to review mine or anybody package to be added to the QubesOS-contrib repository or I develop them by myself. So far, I've been doing the latter, but for bigger projects such as split-dm-crypt or split-browser, I won't copy it because they require quite some work and upstream is responsive.

I will probably finish the screenshot script and add the option to add the contrib repos later on, as it might benefit in the future users who use contributed packages.

**Comment by ben-grande** [comment]
> The main thing here is really to have contrib repos activated alongside all qusal helpers (cacher) and then deploy that tool as is + keyboard shortcut remappings (xfce and KDE).

**Comment by tlaurion** [comment]
@ben-grande 

> reopening until it is done by contribution, as I couldn't do it now. Probably need to look at qdbus and kglobalaccel

Was wrongly closed by rewritting git history in last commits? (should review other issues maybe fitting in that time-window)

Will try to allocate time into testing this (in my todos)

**Comment by tlaurion** [comment]
Hmmmm. sems like BOOTSTRAPP.md should place dev missing requirement between dom0 [...] and sys-cacher (otherwise dotfiles-copy-x11-home and dotfiles-copy-x11-skel are expected to be deployed but aren't?)

**Comment by ben-grande** [comment]
> @ben-grande
> 
> > reopening until it is done by contribution, as I couldn't do it now. Probably need to look at qdbus and kglobalaccel
> 
> Was wrongly closed by rewritting git history in last commits? (should review other issues maybe fitting in that time-window)

Yes, due to [removing mirage tarball retroactively](https://github.com/ben-grande/qusal/issues/32#issuecomment-2007935619).

> Will try to allocate time into testing this (in my todos)

Thanks.

**Comment by ben-grande** [comment]
> Hmmmm. sems like BOOTSTRAPP.md should place dev missing requirement between dom0 [...] and sys-cacher (otherwise dotfiles-copy-x11-home and dotfiles-copy-x11-skel are expected to be deployed but aren't?)

Can you please clarify what *dev missing requirement between dom0 [...] and sys-cacher* means? I see `sys-cacher.configure.sls` with `dotfiles.copy-x11` included.

**Comment by tlaurion** [comment]
> > Hmmmm. sems like BOOTSTRAPP.md should place dev missing requirement between dom0 [...] and sys-cacher (otherwise dotfiles-copy-x11-home and dotfiles-copy-x11-skel are expected to be deployed but aren't?)
> 
> Can you please clarify what _dev missing requirement between dom0 [...] and sys-cacher_ means? I see `sys-cacher.configure.sls` with `dotfiles.copy-x11` included.

Disregard.

deployed salt recipes were not in sync (setup was not ran after git repo copied over dom0...)

**Comment by tlaurion** [comment]
Cross-referencing to upstream https://github.com/QubesOS/qubes-issues/issues/953#issuecomment-2057613527

**Comment by ben-grande** [comment]
If [upstream accepts me as the new maintainer](https://github.com/QubesOS/qubes-issues/issues/953#issuecomment-2371400981), this issue can be closed and KDE shortcuts issue will be redirected to upstream ticketing system.

**Comment by ben-grande** [comment]
I still can't configure KDE shortcuts via the command-line. This is not something I plan to spend more time on. Screenshot tool was implemented and even tried to add to upstream. Closing as won't do KDE shortcuts.
